tests/common/secrets.rs

```diff
@@ -0,0 +1,137 @@
+use anyhow::{Context, Result};
+use async_nats::Client;
+use std::{collections::HashSet, process::Output, time::Duration};
+
+use wasmcloud_secrets_types::Secret;
+
+use super::BackgroundServer;
+
+#[derive(Debug)]
+pub struct NatsKvSecretsBackend {
+    encryption_xkey: nkeys::XKey,
+    transit_xkey: nkeys::XKey,
+    subject_base: String,
+    secrets_bucket: String,
+    nats_address: String,
+    nats_client: Client,
+}
+
+impl NatsKvSecretsBackend {
+    pub async fn new(
+        subject_base: String,
+        secrets_bucket: String,
+        nats_address: String,
+    ) -> Result<Self> {
+        Ok(Self {
+            encryption_xkey: nkeys::XKey::new(),
+            transit_xkey: nkeys::XKey::new(),
+            subject_base,
+            secrets_bucket,
+            nats_client: async_nats::connect(nats_address.clone()).await?,
+            nats_address,
+        })
+    }
+
+    pub async fn ensure_build(&self) -> Result<Output> {
+        std::env::set_current_dir("crates/secrets-nats-kv")?;
+        let res = tokio::process::Command::new("cargo")
+            .arg("build")
+            .output()
+            .await;
+        std::env::set_current_dir("../../")?;
+        res.map_err(|e| anyhow::anyhow!(e))
+    }
+
+    pub async fn put_secret(&self, secret: Secret) -> Result<()> {
+        let request_xkey = nkeys::XKey::new();
+        let mut headers = async_nats::HeaderMap::new();
+        headers.insert(
+            wasmcloud_secrets_types::WASMCLOUD_HOST_XKEY,
+            request_xkey
+                .public_key()
+                .parse::<async_nats::HeaderValue>()
+                .unwrap(),
+        );
+
+        // NOTE: This is just demonstrative that you only need the public key to seal the secret
+        let transit_xkey_pub = nkeys::XKey::from_public_key(&self.transit_xkey.public_key())
+            .expect("public key to be valid");
```

Comment:
I mean, it could basically be a safety comment given we just generated it, but in a test I wasn't really worried about this panic-ing

Reply:
Generally my rule of thumb is if a test function returns a Result to not panic, otherwise, panicking is fine. This is more of a nit for me though in this specific case